File: cdl/src/java/other/org/himalay/cl/cdl/ResourceHeader.java
// Copyright (2013) Krishna C Tripathi. All rights reserved.
// 
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
package org.himalay.cl.cdl ;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.lang.Exception;
import java.io.ByteArrayInputStream;
import java.io.PrintStream;
import org.himalay.msgs.runtime.Created;

import org.himalay.msgs.runtime.*;
@Created(date = "Fri Jan 15 01:39:12 EST 2016")

public  class ResourceHeader extends BinStruct implements PublicBinMsg{




    // members variables
    // messageType
    public short messageType ;
    // descriptor_len
    public short descriptor_len ;

        public ResourceHeader () // throws Exception
    {
            init();
    }
    
    private void init()
    {
        // Initialize messageType
        
        // Initialize descriptor_len
        
    }

    public int readNoHeader(DataInputStream istream) throws IOException 
    {
    
     return read(istream);
       }
   
    public int read(DataInputStream istream) throws IOException 
    {
    preRead();
        int retVal= 0;

                     // read messageType
        {messageType=(short)(istream.readUnsignedByte()); retVal+=1;}
        // read descriptor_len
        {descriptor_len=(short)(istream.readUnsignedByte()); retVal+=1;}

        postRead();
        return retVal;
    }


    public int write(DataOutputStream ostream) throws IOException 
    {
    preWrite();
        int retVal= 0;

        
        
    
                // write messageType
        ostream.writeByte(messageType); retVal +=1;
        // write descriptor_len
        ostream.writeByte(descriptor_len); retVal +=1;
postWrite();
        return retVal;
    }
    
    public int dump(DumpContext dc) throws IOException 
    {
        dc.indent();dc.getPs().print("ResourceHeader\n");
    dc.increaseIndent();
        int retVal= 0;
        // write messageType
        dc.indent();dc.getPs().println("messageType="+messageType+"(0x"+ Integer.toHexString(messageType) + ")" );
        // write descriptor_len
        dc.indent();dc.getPs().println("descriptor_len="+descriptor_len+"(0x"+ Integer.toHexString(descriptor_len) + ")" );
dc.decreaseIndent();
        return retVal;
    }


        // Getter for messageType
    public short getMessageType()
    {
        return messageType ;
    }

    
    // Setter for messageType
    public void setMessageType(short val)
    {
        this.messageType= val;
    }
    // Getter for descriptor_len
    //public short getDescriptor_len()
    //{
    //    return descriptor_len ;
    //}

    
    // Setter for descriptor_len
    //public void setDescriptor_len(short val)
    //{
    //    this.descriptor_len= val;
    //}


    public int getSize() throws IOException
    {
       DataOutputStream dos= new DataOutputStream(new NullStream());
       return this.write(dos);
    }
    
        

}

// End of code

File: libcdl/src/bp/ResourceHeader.cpp
//**//
//@Created(date = "Mon Feb 01 22:30:22 EST 2016")
// Copyright (2012) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
//**//
#include "stdio.h"
#include "ResourceHeader.h"

#ifdef org_himalay_cl_cdl_ResourceHeader__USE_SMART_PTR
#define ArrayList(x) SmartPtrList<x>
#define ArrayList_iterator(x) SmartPtrList<x>::iterator
#include "SmartPtrList.h"
#else
#define ArrayList(x) BinMessagePtrList
#define ArrayList_iterator(x) BinMessagePtrList::iterator
#include "BinMessagePtrList.h"
#endif

// Namespace
namespace org {
namespace himalay {
namespace cl {
namespace cdl {

using namespace ::org::himalay::msgs::runtime;

// Constructor
ResourceHeader::ResourceHeader() {
    references= 0;


    // messageType
    messageType= (ui8)0;
    // descriptor_len
    descriptor_len= (ui8)0;
}
// Destructor
ResourceHeader::~ResourceHeader() {
    // messageType

    // descriptor_len

}

int ResourceHeader::readNoHeader(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;

    // messageType
    {
        messageType=(istream.readUI8());
        retVal+=1;
    }
    // descriptor_len
    {
        descriptor_len=(istream.readUI8());
        retVal+=1;
    }
    return retVal;
}

int ResourceHeader::read(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;
    // read messageType
    {
        messageType=(istream.readUI8());
        retVal+=1;
    }
    // read descriptor_len
    {
        descriptor_len=(istream.readUI8());
        retVal+=1;
    }

    return retVal;
}

int ResourceHeader::write(DataOutputStream& ostream) { // throws IOException



    int retVal= 0;
    // write messageType
    ostream.writeUI8(messageType);
    retVal +=1;
    // write descriptor_len
    ostream.writeUI8(descriptor_len);
    retVal +=1;

    return retVal;
}

int ResourceHeader::dump(DumpContext& dc) { // throws IOException
    dc.indent();
    dc.getPs().print("ResourceHeader\n");
    dc.increaseIndent();
    int retVal= 0;
    // write messageType
    dc.indent();
    dc.getPs().print("messageType=");
    dc.getPs().println((long)messageType);
    // write descriptor_len
    dc.indent();
    dc.getPs().print("descriptor_len=");
    dc.getPs().println((long)descriptor_len);
    dc.decreaseIndent();
    return retVal;
}


// Getter for messageType
ui8 ResourceHeader::getMessageType()
{
    return this->messageType;
}

// Setter for messageType
void ResourceHeader::setMessageType(ui8 val)
{
    this->messageType= val;
}

// Getter for descriptor_len
//ui8 ResourceHeader::getDescriptor_len()
//{
//return this->descriptor_len;
//}

// Setter for descriptor_len
//void ResourceHeader::setDescriptor_len(ui8 val)
//{
//this->descriptor_len= val;
//}

int ResourceHeader::getSize() { // throws IOException
    int retVal = 0;

    return retVal;
}

#ifdef org_himalay_cl_cdl_ResourceHeader__USE_SMART_PTR
void intrusive_ptr_add_ref(org::himalay::cl::cdl::ResourceHeader* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::cl::cdl::ResourceHeader* p) {
    p->decreaseRef();
};;
#endif


}
}
}
}

// End of code